ts/session/apis/session_backend_server.ts

```diff
@@ -0,0 +1,227 @@
+  readonly server: SessionServerConfigType;
+
+  constructor(server: Omit<SessionServerConfigType, 'timeoutMs'>, timeoutMsOverride?: number) {
+    this.server = {
+      ...server,
+      timeoutMs: timeoutMsOverride || 10 * DURATION.SECONDS,
+    };
+  }
+
+  private static getGenericErrorResponse() {
+    return {
+      status_code: 500,
+      t: NetworkTime.nowSeconds(),
+    };
+  }
+
+  public logInfo(msg: string, path?: string) {
+    window.log.info(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}${msg}`);
+  }
+  public logError(msg: string, path?: string) {
+    window.log.error(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}${msg}`);
+  }
+  public logZodError(zodError: ZodError, path?: string) {
+    window.log.error(`[${this.server.name}] ${path?.length ? `${path}: ` : ''}`, zodError);
+  }
+
+  // NOTE: returns null if the ed25519 key pair is not found
+  private async getBlindSignedHeaders({
+    method,
+    path,
+    body,
+  }: {
+    method: string;
+    path: string;
+    body: string | null;
+  }): Promise<BlindSignedHeaders | null> {
+    const ed25519SecretKey = (await UserUtils.getUserED25519KeyPairBytes()).privKeyBytes;
+    if (!ed25519SecretKey) {
+      this.logError(`${path} error: ed25519Secret key was not found when creating headers!`);
+      return null;
+    }
+
+    const blindedPkHex = await BlindingActions.blindVersionPubkey({
+      ed25519SecretKey,
+    });
+
+    const sigTimestampSeconds = NetworkTime.nowSeconds();
+    const signedData = await BlindingActions.blindVersionSignRequest({
+      ed25519SecretKey,
+      sigTimestampSeconds,
+      sigMethod: method,
+      sigPath: path,
+      sigBody: body ? new TextEncoder().encode(body) : null,
+    });
+
+    return {
+      'X-FS-Pubkey': blindedPkHex,
+      'X-FS-Timestamp': sigTimestampSeconds.toString(),
+      'X-FS-Signature': fromUInt8ArrayToBase64(signedData),
+    };
+  }
+
+  private async _makeRequest({
+    path,
+    method,
+    bodyGetter,
+    blindSignRequest,
+  }: SessionBackendServerMakeRequestParams): Promise<SessionBackendServerApiResponse> {
+    const body = typeof bodyGetter === 'function' ? await bodyGetter() : null;
+
+    const headers = blindSignRequest
+      ? await this.getBlindSignedHeaders({
+          method,
+          path,
+          body,
+        })
+      : {};
+
+    if (headers === null) {
+      this.logError('failed to blind sign request parameters', path);
+      return SessionBackendServerApi.getGenericErrorResponse();
+    }
+
+    const url = new URL(path, this.server.url);
+
+    const controller = new AbortController();
+    const result = await timeoutWithAbort(
+      OnionSending.sendViaOnionV4ToNonSnodeWithRetries(
+        this.server.xPkHex,
+        url,
+        {
+          method,
+          headers,
+          body,
+          useV4: true,
+        },
+        false,
+        controller.signal,
+        this.server.timeoutMs
+      ),
+      this.server.timeoutMs,
```

Comment:
I think this needs to be tweaked as we want the inner timeout and outer timeout different, from memory